xml/ha_cluster_lvm.xml

```diff
@@ -258,18 +258,47 @@
       </listitem>
       <listitem>
        <para>
-        Use shared activation mode for &ocfs; and add it to the cloned
-        <literal>g-storage</literal> group:
+        Use shared activation mode for &ocfs;:
        </para>
 <screen>&prompt.root;<command>crm configure primitive vg1 LVM-activate \
    params vgname=vg1 vg_access_mode=lvmlockd activation_mode=shared \
    op start timeout=90s interval=0 \
    op stop timeout=90s interval=0 \
-   op monitor interval=30s timeout=90s</command>
-&prompt.root;<command>crm configure modgroup g-storage add vg1</command></screen>
+   op monitor interval=30s timeout=90s</command></screen>
       </listitem>
      </itemizedlist>
     </step>
+    <step>
+      <para>
+        Make sure the VG can only be activated on nodes where the DLM and
+        <systemitem>lvmlockd</systemitem> resources are already running:
+      </para>
+      <itemizedlist>
+        <listitem>
+          <para>
+            For a single VG using shared activation mode, you can add the VG
+            to the cloned <literal>g-storage</literal> group, which already has
+            internal colocation and order constraints:
+          </para>
+<screen>&prompt.root;<command>crm configure modgroup g-storage add vg1</command></screen>
+        </listitem>
+        <listitem>
+          <para>
+            For a single VG using exclusive activation mode, add constraints directly:
+          </para>
+<screen>&prompt.root;<command>crm configure colocation col-vg-with-dlm inf: vg1 cl-storage</command>
+&prompt.root;<command>crm configure order o-dlm-before-vg Mandatory: cl-storage vg1</command></screen>
+        </listitem>
+        <listitem>
+          <para>
+            For multiple VGs using either activation mode, you can add constraints
+            that apply to batches of resources:
```

Comment:
The example in below looks like it's exclusive activation mode, given that vg1 and vg2 are not explicitly cloned here. How about being more clear, for example something like:

"For multiple VGs, you can add constraints that apply to batches of resource. For instance for exclusive activation mode:
..."

And after the example:
"For multiple VGs using shared activation mode, you can first clone vg1 and vg2, and add the similar constraints that apply to the clone resources instead".

Just an idea, I believe there are other ways to clarify it :-)

Reply:
Cool. Thx.

Reply:
Ah, thank you, I didn't catch that because the original procedure didn't have a step for cloning, but that's because it added the shared mode VG to the group, and the group is cloned. So of course without the group you have to clone. I'll add another dot point to address this.